### ISSUE TITLE
[stable/jenkins] Config-reload problem detection is broken

### ISSUE THREAD

**Issue by holmesb** [helm/charts#18961]
@torstenwalter's [PR#18400](https://github.com/helm/charts/pull/18400) switched the sidecar docker image used for CasC reload.  Kiwigrid's lacks CasC failure-detection.  With mine ([shadwell/k8s-sidecar](https://hub.docker.com/r/shadwell/k8s-sidecar)), [a boolean](https://github.com/holmesb/k8s-sidecar/blob/master/sidecar/sidecar.py#L30) tells if reloading has been successful.  See [feature description under "Logging and Security" here](https://hub.docker.com/r/shadwell/k8s-sidecar).     If reloading was not successful, a pipeline (eg staging followed by prod Jenkins) can be halted to prevent problems with production Jenkins instance.

Can't just reinstate this functionality by switching back the [docker image](https://github.com/helm/charts/blob/ec61610d90db7ffb005111100554dd713196ee06/stable/jenkins/values.yaml#L227) because the supporting SSH config has been removed.

Two solutions:
1. (preferable) Give some indication that can be easily read by a CI\CD script, whether JCasC reload using the new REST method has been successful.
2.  If 1 won't happen soon, reinstate the SSH config so that shadwell/k8s-sidecar can be restored and the stability of prod Jenkins instances isn't put at risk due to bad config.

**Comment by holmesb** [comment]
As a secondary issue, is it possible (as with shadwell/k8s-sidecar) to view logs showing reload status anymore?

**Comment by holmesb** [comment]
Just realised I should be able to fork kiwigrid's sidecar again, include my failure-detection and logging but exclude the SSH authn.  I'll close this for the time being.

**Comment by holmesb** [comment]
For anyone else who uses Configuration as Code validation, I've released v0.0.4 of shadwell/k8s-sidecar:0.0.4.  I've removed the SSH reload method, so it's aligned with kiwigrid/k8s-sidecar except:
- Jenkins CasC validation boolean output to text file (/app/reload_successful.txt)
- Outputs timestamped, categorised Python logs instead of print statements
- Jenkins user created and files re-owned to it.  Root user is not used for increased security.

To use, simply set this value: `master.sidecars.configAutoReload.image: shadwell/k8s-sidecar:0.0.4`
